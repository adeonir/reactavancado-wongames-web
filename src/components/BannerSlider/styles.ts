import { BannerContainer } from 'components/Banner/styles'
import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const BannerSliderContainer = styled.div`
  ${({ theme }) => css`
    .slick-dots {
      list-style: none;
      display: flex !important;
      align-items: center;
      justify-content: center;
      margin-top: ${theme.spacings.small};

      li {
        background: ${theme.colors.white};
        margin: 0 ${theme.spacings.xxsmall};
        width: 1.2rem;
        height: 1.2rem;
        border-radius: 100%;
        display: flex;
        align-items: center;
        justify-content: center;
        cursor: pointer;

        &.slick-active {
          background: ${theme.colors.primary};
        }
      }
      button {
        width: 1.2rem;
        height: 1.2rem;
        opacity: 0;
        cursor: pointer;
      }
    }
    ${media.greaterThan('large')`
      ${BannerContainer} {
        max-width: 104rem;
        margin: 0 auto;
      }
      .slick-dots {
        flex-direction: column;
        height: 100%;
        position: absolute;
        top: 0;
        right: 0;
        margin: 0;

        li {
          margin: ${theme.spacings.xxsmall} 0;
        }
      }
    `}
  `}
`
