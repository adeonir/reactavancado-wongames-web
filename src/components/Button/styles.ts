import styled, { css, DefaultTheme } from 'styled-components'

import { ButtonProps } from '.'

type ContainerProps = {
  hasIcon: boolean
} & Pick<ButtonProps, 'size' | 'fullWidth'>

const modifiers = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xsmall};
    padding: ${theme.spacings.xxsmall} ${theme.spacings.xsmall};
    height: 3rem;

    svg {
      width: ${theme.font.sizes.small};

      & + span {
        margin-left: ${theme.spacings.xxsmall};
      }
    }
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacings.xxsmall} ${theme.spacings.small};
    height: 4rem;

    svg {
      width: ${theme.font.sizes.medium};
    }
  `,
  large: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
    padding: ${theme.spacings.xxsmall} ${theme.spacings.xlarge};
    height: 5rem;

    svg {
      width: ${theme.font.sizes.large};
    }
  `,
  fullWidth: () => css`
    width: 100%;
  `,
  withIcon: (theme: DefaultTheme) => css`
    svg + span {
      margin-left: ${theme.spacings.xxsmall};
    }
  `,
}

export const ButtonContainer = styled.button<ContainerProps>`
  ${({ theme, size, fullWidth, hasIcon }) => css`
    background-image: linear-gradient(180deg, #ff5f5f 0%, #f062c0 50%);
    color: ${theme.colors.white};
    border-radius: ${theme.border.radius};
    border: 0;
    cursor: pointer;
    position: relative;
    z-index: ${theme.layers.base};
    text-decoration: none;

    display: inline-flex;
    align-items: center;
    justify-content: center;

    &:before {
      background-image: linear-gradient(180deg, #e35565 0%, #d958a6 50%);
      content: '';
      border-radius: inherit;
      display: block;
      position: absolute;
      top: 0;
      right: 0;
      bottom: 0;
      left: 0;
      z-index: -1;
      opacity: 0;
      transition: opacity 0.2s ease;
    }

    &:hover {
      &:before {
        opacity: 1;
      }
    }

    ${!!size && modifiers[size](theme)}
    ${!!fullWidth && modifiers.fullWidth}
    ${!!hasIcon && modifiers.withIcon(theme)}
  `};
`
