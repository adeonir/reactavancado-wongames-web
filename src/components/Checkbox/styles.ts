import styled, { css } from 'styled-components'

import { CheckboxProps } from '.'

export const CheckboxContainer = styled.div`
  display: flex;
  align-items: center;
`

export const Input = styled.input`
  ${({ theme }) => css`
    cursor: pointer;
    display: flex;
    align-items: center;
    justify-content: center;
    appearance: none;
    width: 1.8rem;
    height: 1.8rem;
    border-radius: 0.2rem;
    border: 0.2rem solid ${theme.colors.gray};
    position: relative;
    transition: background border ${theme.transition.fast};

    &::before {
      content: '';
      width: 0.6rem;
      height: 0.9rem;
      border: 0.2rem solid ${theme.colors.white};
      border-top: 0;
      border-left: 0;
      position: absolute;
      top: 0.1rem;
      transform: rotate(45deg);
      transition: background border ${theme.transition.fast};
      opacity: 0;
    }

    &:focus {
      box-shadow: 0 0 0.5rem ${theme.colors.primary};
    }

    &:checked {
      background: ${theme.colors.primary};
      border-color: ${theme.colors.primary};

      &::before {
        opacity: 1;
      }
    }
  `}
`

export const Label = styled.label<Pick<CheckboxProps, 'labelColor'>>`
  ${({ theme, labelColor }) => css`
    color: ${theme.colors[labelColor!]};
    cursor: pointer;
    padding-left: ${theme.spacings.xxsmall};
    line-height: 1;
  `}
`
