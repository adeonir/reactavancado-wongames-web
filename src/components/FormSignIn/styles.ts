import styled from 'styled-components'

export const FormSignInContainer = styled.div``

export const ForgotPassword = styled.a``

export const FormLink = styled.div``
