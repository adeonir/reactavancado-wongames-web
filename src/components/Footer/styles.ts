import { HeadingContainer } from 'components/Heading/styles'
import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const FooterContainer = styled.footer`
  ${HeadingContainer} {
    text-transform: uppercase;
  }
`

export const Content = styled.div`
  ${({ theme }) => css`
    display: grid;
    gap: ${theme.grid.gutter};
    margin-top: ${theme.spacings.medium};
    grid-template-columns: repeat(2, 1fr);

    ${media.greaterThan('medium')`
    grid-template-columns: repeat(4, 1fr);
    `}
  `}
`

export const Column = styled.div`
  ${({ theme }) => css`
    a,
    span {
      display: block;
      color: ${theme.colors.gray};
      margin-bottom: ${theme.spacings.xxsmall};
      font-size: ${theme.font.sizes.small};
      text-decoration: none;
    }

    a:hover {
      text-decoration: underline;
    }
  `}
`

export const Copy = styled.div`
  ${({ theme }) => css`
    color: ${theme.colors.gray};
    margin-top: ${theme.spacings.large};
    margin-bottom: ${theme.spacings.medium};
    font-size: ${theme.font.sizes.xsmall};
    text-align: center;
  `}
`
