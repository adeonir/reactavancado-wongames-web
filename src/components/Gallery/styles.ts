import { rgba } from 'polished'
import styled, { css } from 'styled-components'
import media from 'styled-media-query'

import { theme } from 'styles'

export const GalleryContainer = styled.div`
  .slick-prev,
  .slick-next {
    display: block;
    color: ${theme.colors.white};
    cursor: pointer;
    position: absolute;
    top: 50%;
    width: 2.5rem;
    height: 2.5rem;
    padding: 0;
    transform: translate(0, -50%);
  }

  .slick-prev {
    left: -${theme.spacings.xxlarge};
  }

  .slick-next {
    right: -${theme.spacings.xxlarge};
  }

  .slick-prev.slick-disabled,
  .slick-next.slick-disabled {
    visibility: hidden;
  }

  .slick-slide > div {
    margin: 0 ${theme.spacings.xsmall};
    cursor: pointer;
  }

  .slick-list {
    margin: 0 -${theme.spacings.xsmall};
  }

  ${media.lessThan('huge')`
    overflow-x: hidden;
  `}
`

type ModalProps = {
  isOpen: boolean
}

const modifiers = {
  open: () => css`
    opacity: 1;
  `,
  close: () => css`
    opacity: 0;
    pointer-events: none;
  `,
}

export const Modal = styled.div<ModalProps>`
  ${({ isOpen }) => css`
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    background: ${rgba(theme.colors.black, 0.5)};
    backdrop-filter: blur(5px);
    display: flex;
    align-items: center;
    justify-content: center;
    index: ${theme.layers.modal};
    transition: opacity ${theme.transition.default};

    ${isOpen && modifiers.open}
    ${!isOpen && modifiers.close}
  `}
`

export const Close = styled.div`
  color: ${theme.colors.white};
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  cursor: pointer;
  padding: ${theme.spacings.xsmall};
  text-align: right;
`
