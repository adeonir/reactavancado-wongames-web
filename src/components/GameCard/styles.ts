import { shade, tint } from 'polished'
import styled, { css, DefaultTheme } from 'styled-components'

import { theme } from 'styles'

type PriceProps = {
  isPromotional?: boolean
}

const modifiers = {
  default: (theme: DefaultTheme) => css`
    border-radius: ${theme.border.radius};
    color: ${theme.colors.white};
    background: ${theme.colors.secondary};
    padding: 0 ${theme.spacings.xxsmall};
    margin-right: calc(${theme.spacings.xxsmall} / 2);
  `,
  promotional: (theme: DefaultTheme) => css`
    color: ${theme.colors.gray};
    margin-right: ${theme.spacings.xsmall};
    text-decoration: line-through;
  `,
}

export const GameCardContainer = styled.article`
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  background-color: ${theme.colors.white};
  position: relative;
`

export const ImageBox = styled.div`
  width: 100%;
  height: 14rem;
  background: ${shade(0.02, theme.colors.white)};
  background-image: linear-gradient(
    to right,
    ${shade(0.02, theme.colors.white)} 0%,
    ${tint(0.78, theme.colors.secondary)} 20%,
    ${shade(0.02, theme.colors.white)} 40%,
    ${shade(0.02, theme.colors.white)} 100%
  );
  background-size: 80rem 14rem;
  animation: shimmer 1s linear infinite forwards;

  img {
    width: 100%;
    height: 100%;
    object-fit: cover;
  }

  @keyframes shimmer {
    0% {
      background-position: -40rem 0;
    }
    100% {
      background-position: 40rem 0;
    }
  }
`

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  position: relative;
  height: 100%;
  margin: ${theme.spacings.xsmall};
`

export const Info = styled.div`
  max-width: calc(100% - 2.5rem);
`

export const Title = styled.h3`
  color: ${theme.colors.black};
  font-size: ${theme.font.sizes.medium};
  line-height: ${theme.font.sizes.medium};
  font-weight: ${theme.font.bold};
`

export const Developer = styled.h4`
  color: ${theme.colors.gray};
  font-size: ${theme.font.sizes.small};
  font-weight: ${theme.font.normal};
`

export const FavButton = styled.div`
  color: ${theme.colors.primary};
  cursor: pointer;
  position: absolute;
  right: 0;
  top: -0.5rem;

  svg {
    width: 2.5rem;
  }
`

export const PriceBox = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-end;
  margin-top: ${theme.spacings.xxsmall};
`

export const Price = styled.div<PriceProps>`
  ${({ isPromotional }) => css`
    display: inline-flex;
    align-items: center;
    font-weight: ${theme.font.bold};
    height: 3rem;

    ${!isPromotional && modifiers.default(theme)}
    ${isPromotional && modifiers.promotional(theme)}
  `};
`
