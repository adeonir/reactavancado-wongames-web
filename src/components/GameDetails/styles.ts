import styled from 'styled-components'
import media from 'styled-media-query'

import { theme } from 'styles'

export const GameDetailsContainer = styled.div`
  margin: ${theme.spacings.small} 0;
`

export const Content = styled.div`
  display: grid;
  gap: ${theme.spacings.xsmall};
  grid-template-columns: repeat(2, 1fr);
  margin-top: ${theme.spacings.small};
  ${media.greaterThan('medium')`
    grid-template-columns: repeat(3, 1fr);
  `}
  ${media.greaterThan('large')`
    grid-template-columns: repeat(6, 1fr);
  `}
`

export const Label = styled.h3`
  font-size: ${theme.font.sizes.small};
  font-weight: ${theme.font.light};
  color: ${theme.colors.lightGray};
`

export const Title = styled.p`
  font-size: ${theme.font.sizes.medium};
  font-weight: ${theme.font.bold};
  color: ${theme.colors.white};
`

export const IconGroup = styled.div`
  color: ${theme.colors.white};
`

export const Icon = styled.span`
  margin-right: ${theme.spacings.xxsmall};
`
