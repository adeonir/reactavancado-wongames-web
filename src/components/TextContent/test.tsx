import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import { theme } from 'styles'

import { TextContent } from '.'

const props = {
  title: 'Description',
  content: `<h1>Content</h1>`,
}

describe('<TextContent />', () => {
  it('should render the title and content', () => {
    renderWithTheme(<TextContent {...props} />)

    expect(
      screen.getByRole('heading', { name: /description/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /content/i })
    ).toBeInTheDocument()
  })

  it('should render without the title', () => {
    renderWithTheme(<TextContent content={props.content} />)

    expect(
      screen.queryByRole('heading', { name: /description/i })
    ).not.toBeInTheDocument()
  })

  it('should render text in white when smaller then 768px', () => {
    renderWithTheme(<TextContent {...props} />)

    const container = screen.queryByRole('heading', { name: /description/i })
      ?.parentElement

    expect(container).toHaveStyle({
      color: theme.colors.white,
    })

    expect(container).toHaveStyleRule('color', theme.colors.black, {
      media: '(min-width: 768px)',
    })
  })

  it('should render background in blue when smaller then 768px', () => {
    renderWithTheme(<TextContent {...props} />)

    const container = screen.queryByRole('heading', { name: /description/i })
      ?.parentElement

    expect(container).toHaveStyleRule('background', theme.colors.white, {
      media: '(min-width: 768px)',
    })
  })
})
