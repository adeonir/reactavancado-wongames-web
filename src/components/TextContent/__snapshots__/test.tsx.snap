// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextContent /> should render the heading 1`] = `
<div
  class=""
>
  <h1>
    TextContent
  </h1>
</div>
`;
